mesonbuild/environment.py

```diff
@@ -913,6 +913,9 @@ def get_build_dir(self) -> str:
 
     def get_import_lib_dir(self) -> str:
         "Install dir for the import library (library used for linking)"
+        m = self.machines.host
```

Comment:
Is it really safe to assume `host` here, or should this be `self.machines[self.for_machine]`?

Reply:
It is the same as code after that. What is the difference? In theory it should be used only when installing libraries that will run on Haiku.

Reply:
We currently do this for windows checks too, yeah.

At the moment this is fine since you can only install host outputs. Changing that is tied to reworking meson for the use case of supporting projects such as QEMU that install for many different toolchains unrelated to the build machine, which will anyways require  reworking all for_machine logic and this won't make it any worse.